src/ResourceManager/Media/Commands.Media/MediaService/NewAzureRmMediaService.cs

```diff
@@ -0,0 +1,176 @@
+    {
+        protected const string PrimaryStorageAccountParamSet = "StorageAccountIdParamSet";
+        protected const string StorageAccountsParamSet = "StorageAccountsParamSet";
+        private const string NewMediaServiceWhatIfMessage = "New a MediaService";
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = PrimaryStorageAccountParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = StorageAccountsParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = PrimaryStorageAccountParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The media service account name.")]
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = StorageAccountsParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The media service account name.")]
+        [ValidateNotNullOrEmpty]
+        [ValidateLength(MediaServiceAccountNameMinLength, MediaServiceAccountNameMaxLength)]
+        [ValidatePattern(MediaServiceAccountNamePattern)]
+        [Alias("Name", "ResourceName")]
+        public string AccountName { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ParameterSetName = PrimaryStorageAccountParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The location.")]
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ParameterSetName = StorageAccountsParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The location.")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(
+            Position = 3,
+            Mandatory = true,
+            ParameterSetName = PrimaryStorageAccountParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The primary storage account assosiated with the media account.")]
+        [Alias("Id")]
+        [ValidateNotNullOrEmpty]
+        public string StorageAccountId { get; set; }
+
+        [Parameter(
+            Position = 3,
+            Mandatory = true,
+            ParameterSetName = StorageAccountsParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The storage accounts assosiated with the media account.")]
+        [ValidateNotNullOrEmpty]
+        public PSStorageAccount[] StorageAccounts { get; set; }
```

Comment:
>PSStorageAccount [](start = 15, length = 16)

1. How do end users will create this type?
2. You do not have test coverage for this one, please add.